subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/StaleDataCleanupInitializer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.results;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class StaleDataCleanupInitializer implements ConnectionAction<Void> {
+    private static final int MAX_DATA_STORE_DAYS = 365;
+
+    @Override
+    public Void execute(Connection connection) throws SQLException {
+        Statement statement = connection.createStatement();
+
+        String startDate = LocalDateTime.now().minusDays(MAX_DATA_STORE_DAYS).format(DateTimeFormatter.ISO_DATE);
+        statement.execute(String.format("delete from testExecution where startTime < '%s'", startDate));
+        statement.execute(String.format("delete from testOperation where testExecution in (select id from testExecution where startTime < '%s')", startDate));
```

Comment:
I'm a little concerned about the performance of `delete from testOperation where testExecution in (select id from testExecution where startTime < '%s')` because the list might be over 10,000 long.

Reply:
I think the easiest way would be to add an `ON DELETE CASCADE` action to foreign key. Then the child entries would be deleted automatically. I don't think h2 has a big problem with a large `IN` where clause, so let's try it out.

Reply:
Sounds good. I was looking for a solution for a few hours, I tried `DELETE JOIN` which works in MySQL but not in H2.